Sources/APIServer/Containers/ContainersService.swift

```diff
@@ -215,33 +215,80 @@ actor ContainersService {
                     message: "container \(id) is not yet stopped and can not be deleted"
                 )
             }
-            try self._cleanup(id: id, item: item)
+            try await self._cleanup(id: id, item: item)
         case .dead, .exited(_):
-            try self._cleanup(id: id, item: item)
+            try await self._cleanup(id: id, item: item)
         }
     }
 
     private static func fullLaunchdServiceLabel(runtimeName: String, instanceId: String) -> String {
         "\(Self.launchdDomainString)/\(Self.machServicePrefix).\(runtimeName).\(instanceId)"
     }
 
-    private func _cleanup(id: String, item: Item) throws {
+    private func _cleanup(id: String, item: Item) async throws {
         self.log.debug("\(#function)")
         let config = try item.bundle.configuration
+
+        // Release volumes before deleting the container
+        do {
+            try await self.releaseVolumes(for: config, containerId: id)
+        } catch {
+            self.log.error("Failed to release volumes for container \(id): \(error)")
+            // Don't fail the cleanup process if volume release fails
+        }
+
         let label = Self.fullLaunchdServiceLabel(runtimeName: config.runtimeHandler, instanceId: id)
         try ServiceManager.deregister(fullServiceLabel: label)
         try item.bundle.delete()
         self.containers.removeValue(forKey: id)
     }
 
+    private func releaseVolumes(for config: ContainerConfiguration, containerId: String) async throws {
+        let volumePrefix = "volumes/"
+        let volumeSuffix = "/volume.img"
+
+        for mount in config.mounts {
+            // Check if this is a named volume by looking at the source path
+            if mount.isBlock && mount.source.contains(volumePrefix) && mount.source.hasSuffix(volumeSuffix) {
+                // Extract volume name from path like ".../volumes/myvolume/volume.img"
+                let sourcePath = mount.source
+                if let volumeStart = sourcePath.range(of: volumePrefix)?.upperBound,
+                    let volumeEnd = sourcePath.range(of: volumeSuffix)?.lowerBound
```

Comment:
This is... crummy that we need to do this. There has to be a better way we can just include at least the volume name in what we send over. The `.block` Filesystem type has everything we need except the volume name which we need to dec the refcount right? I honestly wouldn't be opposed to (I know, we were the ones who told you to use .block) a new FS type that is literally just .block, but with a volume name on included.

Reply:
@jglogan @adityaramani thoughts?